### PR TITLE
CI: Add GitHub action pipeline

### PR DESCRIPTION
**Pull request by evgenykor**
Add github yml configuration to build
setup and test tidesdb

### CODE REVIEW

**Comment by guycipher** [comment]
Hey @evgenykor! Looks awesome.  I have a question with the pipeline does it block PRs and commits that don’t pass?  Do you have docs on this process?  I haven’t used too much of this process and want to understand how the flow works 100%.   Thank you!

**Comment by evgenykor** [comment]
Hi @guycipher  please refer to https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-protected-branches/managing-a-branch-protection-rule 

TLDR: just having this action won't block anything by default - you may freely merge everything you want to

**Comment by guycipher** [comment]
Reviewing.  I saw it builds with Ubuntu it looks fine, generally yes.  I wonder if it’s possible to build windows and macOS probably not!  This is still solid.  Give me couple mins to review quick there.

**Comment by evgenykor** [comment]
Yeah it would be good to investigate Windows pipeline

**Comment by guycipher** [comment]
I think you pasted wrong link

**Comment by guycipher** [comment]
I’ve reviewed.  I like it.  Could you explain couple pieces.  I may have to test it.  In cmake we have build and test flags.  We must with the CI be able to run all tests from CI.  Not always will cmake have test and debug flags ok so this would require some condition logic somehow.  This is where it may get complex a bit.

File: .github/workflows/build_and_test_tidesdb.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    - name: configure cmake build for tidesdb
+      run: | 
+        cd tidesdb
+        cmake --debug-output  -S . -B build && make -C build/
+
+    - name: run tests
+      run: |
+        export ASAN_OPTIONS=verify_asan_link_order=0
+        cd tidesdb/build
+        ctest
```
Comment (evgenykor) [inline .github/workflows/build_and_test_tidesdb.yml:33]: @guycipher this is test command that we run. actually better to run it with `--output-on-failure` flag but anyway. 

so if you want some env variable sets this is good place to set them (via export)

Comment (evgenykor) [reply .github/workflows/build_and_test_tidesdb.yml:33]: it is also possible to set several combinations of env. variables and during build config (line #27) e.t.c.

Comment (guycipher) [reply .github/workflows/build_and_test_tidesdb.yml:33]: Understood.  I see how it works.  I’m going to office now but have you been able to test this or we have to merge first?  I will assist in testing.

Comment (evgenykor) [reply .github/workflows/build_and_test_tidesdb.yml:33]: Well it is partially used in tidesdb-lua action  and that already been tested well. So I doubt it won't work. And yes we have to merge it first for GH in order to test it unfortinately.

Comment (evgenykor) [reply .github/workflows/build_and_test_tidesdb.yml:33]: You always may just delete the file in the following revert commit if you don't like it - it is not a big deal

Comment (guycipher) [reply .github/workflows/build_and_test_tidesdb.yml:33]: No problemo.  We shall test it!

**Pull request merged**